### PR TITLE
[Dashboard] [Serve] Make serve import conditional

### LINKED ISSUE #17712: [serve] fastapi and uvicorn are hard dependencies
### What is the problem?

*Ray version and other system information (Python version, TensorFlow version, OS):*

In the nightly wheels, `fastapi` and `uvicorn` became hard dependencies, causing an exception to be thrown on `ray.init()` if they are not present - and they are not installed by `ray[default]`.

```
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/agent.py", line 380, in <module>
    loop.run_until_complete(agent.run())
  File "/home/ubuntu/anaconda3/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
    return future.result()
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/agent.py", line 161, in run
    modules = self._load_modules()
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/agent.py", line 111, in _load_modules
    dashboard_utils.DashboardAgentModule)
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/utils.py", line 208, in get_all_modules
    importlib.import_module(name)
  File "/home/ubuntu/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/modules/snapshot/snapshot_head.py", line 7, in <module>
    from ray.serve.controller import SNAPSHOT_KEY as SERVE_SNAPSHOT_KEY
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/ray/serve/__init__.py", line 1, in <module>
    from ray.serve.api import (connect, start, get_replica_context, get_handle,
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/ray/serve/api.py", line 15, in <module>
    from starlette.requests import Request
ModuleNotFoundError: No module named 'starlette'
```

(line 16 for `uvicorn`).

### Reproduction (REQUIRED)
Install the latest nightly wheel in a fresh virtualenv:
`pip install -U "ray[default] @ https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-2.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl"`

Run `ray.init()`.

### PR DESCRIPTION
**Pull request by architkulkarni**
## Related issue number

Closes #17712 
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/snapshot/snapshot_head.py [hunk 1]
```diff
@@ -91,6 +88,15 @@ async def get_actor_info(self):
         return actors
 
     async def get_serve_info(self):
+        # Conditionally import serve to prevent ModuleNotFoundError from serve
+        # dependencies when only ray[default] is installed (#17712)
+        try:
+            from ray.serve.controller import SNAPSHOT_KEY as SERVE_SNAPSHOT_KEY
+            from ray.serve.constants import SERVE_CONTROLLER_NAME
+            from ray.serve.kv_store import format_key
+        except Exception:
```
Comment (edoakes) [inline dashboard/modules/snapshot/snapshot_head.py:97]: maybe make this an importerror?

**Review by edoakes — APPROVED** [review]

**Review by ijrsvt — APPROVED** [review]
LGTM!

File: dashboard/modules/snapshot/snapshot_head.py (hunk 1 above)
Comment (wuisawesome) [reply dashboard/modules/snapshot/snapshot_head.py:97]: If we throw something, can we be sure to catch it on the other side? Ideally we should be able to return the core information regardless of whether there a dependency issue with serve.

Comment (architkulkarni) [reply dashboard/modules/snapshot/snapshot_head.py:97]: In the issue it's a ModuleNotFoundError, but I can change it to check for both ImportError and ModuleNotFoundError.

@wuisawesome Sorry, could you spell out what it means to catch it on the other side?  Also, I think this code does still return the core information if serve isn't there, it'll just have `"deployments": {}` in the snapshot.

**Pull request merged**